Src/Support/Google.Apis.Auth/OAuth2/Responses/TokenResponse.cs

```diff
@@ -165,6 +188,12 @@ public static async Task<TokenResponse> FromHttpResponseAsync(HttpResponseMessag
                 // and OIDC token for IAP. The IdToken is used for access in that case.
                 newToken.AccessToken ??= newToken.IdToken;
 
+                // For impersonated credentials access token we need to convert expire time to seconds.
+                if (!String.IsNullOrEmpty(newToken.ImpersonatedAccessTokenExpireTime)) {
+                    var date = DateTime.Parse(newToken.ImpersonatedAccessTokenExpireTime).ToUniversalTime();
```

Comment:
This looks wrong to me - I'd expect it to *already* be in UTC (as per the comment).

We should use `DateTime.ParseExact`, specify the expected format, specify the invariant culture, and `DateTimeStyles.AssumeUniversal | DateTimeStyles.AdjustToUniversal` (I think; I'll need to check that.)

Reply:
The expire time has `"2020-04-07T15:01:23.045123456Z"` format (rfc3339 I guess), so it needs to convert to UTC. I didn't find a working way to parse it using `DateTime.ParseExact`. I tried to use `"yyyy-MM-dd'T'HH:mm:ss'.'fffffffff'Z'"` format but I gave me exceptions.

Reply:
Mmm... the 9 digits causes issues for custom date format strings, it seems. (Wouldn't be a problem for Noda Time...)

I think this would be better though:

```csharp
var expiry = DateTime.Parse(text, CultureInfo.InvariantCulture, DateTimeStyles.AdjustToUniversal);
```

(Note the use of `expiry` rather than `date` as this isn't a date... it's an instant in time.) The main thing is that we never have a local `DateTime` to adjust... we say we want the result of parsing to already be in UTC. We definitely want to pass in the invariant culture though.

Alternatively, and perhaps cleaner:

```csharp
var expiry = DateTimeOffset.Parse(newToken.ImpersonatedAccessTokenExpireTime, Cultureinfo.InvariantCulture);
var timeBeforeExpiry = expiry - clock.UtcNow;
newToken.ExpiresInSeconds = (long) timeBeforeExpiry.TotalSeconds;
```

As still another alternative, we could potentially change `ImpersonatedAccessTokenExpireTime` to be a `DateTimeOffset` and let Json.NET do the parsing. (Then use the last part of the code above.)

Reply:
Updated the code to the first suggested alternative:
```
var expiry = DateTimeOffset.Parse(newToken.ImpersonatedAccessTokenExpireTime, Cultureinfo.InvariantCulture);
var timeBeforeExpiry = expiry - clock.UtcNow;
newToken.ExpiresInSeconds = (long) timeBeforeExpiry.TotalSeconds;
```